carnival.cabal

```diff
@@ -137,9 +139,6 @@ library
                  , yesod-text-markdown
 
 executable         carnival
-    if flag(library-only)
```

Comment:
I like the idea of separate executables. I'm not sure what's wasted in that case, though managing the dependency lists may be a hassle.